File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
  const answer = []
  const days = progresses.map((p, i) => Math.ceil((100 - p) / speeds[i]))
  let prev = 0

  for (let i = 0; i < days.length; i++) {
    if (days[prev] < days[i]) {
      answer.push(i - prev)
      prev = i
    }
  }
  answer.push(days.length - prev)
  return answer
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    answer = []
    days = []
    
    for p, s in zip(progresses, speeds):
        days.append(math.ceil((100 - p) / s))
        
    prev = 0
    
    for i in range(len(days)):
        if days[prev] < days[i]:
            answer.append(i - prev)
            prev = i

    answer.append(len(days) - prev)
    return answer